intelmq/lib/bot_debugger.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""
+Utilities for debugging intelmq bots.
+
+BotDebugger is called via intelmqctl. It starts a live running bot instance,
+leverages logging to DEBUG level and permits even a non-skilled programmer
+who may find themselves puzzled with Python nuances and server deployment twists
+to see what's happening in the bot and where's the error.
+
+Depending on the subcommand received, the class either
+ * starts the bot as is (default)
+ * processes single message, either injected or from default pipeline (process subcommand)
+ * reads the message from input pipeline or send a message to output pipeline (message subcommand)
+"""
+import logging
+import json
+
+from intelmq.lib.message import Event
+from importlib import import_module
+from intelmq.lib.utils import StreamHandler
+from intelmq.lib.message import Event
+
+
+class BotDebugger:
+        def leverageLogger(self, level = logging.DEBUG):
+            self.instance.logger.setLevel(level)
+            for h in self.instance.logger.handlers:
+                if isinstance(h, StreamHandler):
+                    h.setLevel(level)
+
+        def __init__(self, module_path, bot_id, run_subcommand = None, message_kind = None, dryrun = None, msg = None):
+            module = import_module(module_path)
+            bot = getattr(module, 'BOT')
+            self.instance = bot(bot_id)
```

Comment:
This runs the bot's initialization. Thus, if the user only want's to get the message, this should not be run.